### PR TITLE
Add ability to create previews for Hubble Advanced Spectral Products

### PR DESCRIPTION
**Pull request by raswaters**

### CODE REVIEW

File: spec_plots/utils/specutils_hasp/plotspec.py [hunk 1]
```diff
@@ -0,0 +1,347 @@
+"""
+.. module:: plotspec
+   :synopsis: Creates preview plots for the provided HASP spectrum.
+
+.. moduleauthor:: Scott W. Fleming <fleming@stsci.edu>
```
Comment (raswaters) [inline spec_plots/utils/specutils_hasp/plotspec.py:5]: This script is based on the plotspec.py from cos/stis, but was updated significantly, so could update author to Swaters.

File: spec_plots/utils/specutils_hasp/__init__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+"""
+.. module:: __init__
+
+   :synopsis: Used to treat "specutils_stis" as a directory and package.
```
Comment (raswaters) [inline spec_plots/utils/specutils_hasp/__init__.py:4]: Change "stis" to hasp

File: spec_plots/utils/specutils_hasp/plotspec.py [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+
+        if is_bigplot:
+            this_figure.suptitle(os.path.basename(
+                hasp_spectrum.plot_info['orig_file']), fontsize=18, color='r')
+            # Uncomment the lines below to include the covering fraction
+            # as part of the suptitle.
+
+        if debug:
+            # Overplot points color-coded based on rejection criteria.
+            debug_oplot(this_plotarea, "stis", all_wls, all_fls,
```
Comment (raswaters) [inline spec_plots/utils/specutils_hasp/plotspec.py:229]: Change stis to hasp. Verify whether updates to debug_oplot are needed.

File: spec_plots/utils/specutils_hasp/plotspec.py (hunk 1 above)
Comment (raswaters) [reply spec_plots/utils/specutils_hasp/plotspec.py:5]: Updated author to Swaters.

File: spec_plots/utils/specutils_hasp/plotspec.py (hunk 3 above)
Comment (raswaters) [reply spec_plots/utils/specutils_hasp/plotspec.py:229]: Updated "stis" to "hasp". debug_oplot calls is_bd_dpq() and passes in "hasp". is_bad_dpq() can already handle "hasp".

**Comment by scfleming** [comment]
@raswaters Can you bump the version in the `__init__.py` under the main spec_plots folder from '1.35.1' to '1.36.0'?

**Comment by scfleming** [comment]
Can you add a section to the `History.rst` file in the main folder with an entry for a v1.36.0 that states this version supports preview generation of HASP products?

**Comment by scfleming** [comment]
I will do some (by hand) regression testing of regular HST and JWST, and pending no problems there and with the above two items, we should be able to proceed with packaging this new version.

**Comment by scfleming** [comment]
Also looks like we'll need to update some of the Sphinx files in spec_plots/docs/API to include entries for the new hasp modules.  If you want to do this in this PR feel free, if not, I could do it later, but I think I'd want to do a PR of this forked branch you have here into a v1.36.0 branch in the main repo before merging into main branch itself.

**Comment by raswaters** [comment]
I've updated the `__init__.py` and `History.rst` to add the 1.36.0 version. I've also created Sphinx files for the new HASP additions. I've not used Sphinx before, so my additions are entirely based on making some analogous to the same file for COS.

**Pull request merged**